src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIBookingServiceTest.kt

```diff
@@ -43,15 +45,33 @@ internal class CommunityAPIBookingServiceTest {
     val uri = "/appt/X1/123/CRS"
     val link = "http://url/view/${appointment.actionPlan.referral.id}"
     val request = AppointmentCreateRequestDTO("ACC", now, now.plusMinutes(60), now.plusMinutes(120), "CRSEXTL", notes = link, true)
-    val response = AppointmentCreateResponseDTO(1234L)
+    val response = AppointmentResponseDTO(1234L)
 
-    whenever(communityAPIClient.makeSyncPostRequest(uri, request, AppointmentCreateResponseDTO::class.java))
+    whenever(communityAPIClient.makeSyncPostRequest(uri, request, AppointmentResponseDTO::class.java))
       .thenReturn(response)
 
     val deliusAppointmentId = communityAPIBookingService.book(appointment, now.plusMinutes(60), 60)
 
     assertThat(deliusAppointmentId).isEqualTo(1234L)
-    verify(communityAPIClient).makeSyncPostRequest(uri, request, AppointmentCreateResponseDTO::class.java)
+    verify(communityAPIClient).makeSyncPostRequest(uri, request, AppointmentResponseDTO::class.java)
+  }
+
+  @Test
+  fun `requests rescheduling an appointment when timings are different`() {
```

Comment:
would be good to add a test to validate the expected behaviour when the booking is neither 'initial' or 'reschedule'

Reply:
Just added for the above comment:

  @Test
  fun `does not reschedule an appointment when timings are same`